lib/neverblock/core/fiber.rb

```diff
@@ -35,10 +35,88 @@ def self.yield(*args)
   end
 
   private
-  
+
   def local_fiber_variables
     @local_fiber_variables ||= {}
   end
 
 end
 
+# Code that is designed to work with mutexes and threads but isn't aware of fibers
+# will potentially screw up. 
+class Mutex
+  alias :_orig_try_lock :try_lock
+  def try_lock
+    acquired = _orig_try_lock
+    disable_neverblock if acquired
+    acquired
+  end
+
+  # sleep is mostly used by ConditionVariable
+  alias :_orig_sleep :sleep
+  def sleep(*args)
```

Comment:
if sleep is called within a synchronize block, wont this call the same issues again?

Reply:
This is the Mutex sleep not the regular sleep

Reply:
mutex.sleep does the following: 1. unlock the mutex. 2. Calls ruby sleep 3. regets the lock. 2 is usually nil (meaning forever) and its so you can sit and wait for another thread to call wakeup on you -- ConditionVariable uses this function to do just that.

Since it's basically just {self.unlock, sleep, self.lock} think it has to be consistent with what those functions do or I'm not sure what might happen.